### ISSUE TITLE
highlighted variables when focused on a hole should allow the user to click them to insert

### ISSUE THREAD

**Issue by MajeedKazemi** [MajeedKazemi/nova-editor#458]
![image](https://user-images.githubusercontent.com/57059538/139899237-8abef98e-905b-4993-a583-f130bd4617ae.png)

clicking on the green variable should do one of the following:
- instantly insert the variable into the hole
- show a hover message that includes a message on what is going on (e.g. you can insert this variable to that hole because the types match + button to insert)

@vicchig thoughts?

**Comment by vicchig** [comment]
@MajeedKazemi 

I have two issues with this.

1. What would be the indication to a user that they can actually click it and that the corresponding action would insert it somewhere. This is something that is very clear in our toolbox because we have button-looking objects in there, but would be very unclear here.

2. How would we know what context to insert it into? When you are focused on the hole and click the variable, the context changes. We would have to keep the old context and validate the insertion against that I guess. So maybe that is not a problem.

**Comment by MajeedKazemi** [comment]
This came out from a user study that I'm currently doing with a 13 year old student. She simply thought that there is going to be a message (if she hovered over it) or she could click on the green variable. That's why I think it might be a natural interaction

**Comment by vicchig** [comment]
Yeah the message maybe because we have other places that are highlighted and we have messages for those. It is interesting that she thought that it is clickable though because we definitely don't have anything that performs an action that is also directly in the editor window (drafts are outside of it).

Then yeah, if a good amount of users expect, we should include it. I guess it does not hurt for it to be there even if few end up using it. It might be more important to have those who would expect it to be intractable to get the expected result.
